### PR TITLE
3.6.9

### PR DESCRIPTION
**Pull request by sei-bstein**
**Bug fixes**

- Addressed an issue which caused the UI not to properly update when users enrolled or unenrolled from single-player games.
- Resolved an issue that caused repeated API calls and crashes upon selection of a standard (non-Unity) challenge.

### CODE REVIEW

File: projects/gameboard-ui/src/app/game/gameboard-page/gameboard-page.component.ts [hunk 1]
```diff
@@ -139,13 +129,23 @@ export class GameboardPageComponent implements OnDestroy {
...
       this.api.setColor(s);
+
       // TODO: revisit this temp solution for auto-grading sync
-      this.refresh$.next(this.ctx.id);
+      // this.refresh$.next(this.ctx.id);
+
+      // proposed solution for auto-grading sync:
+      this.api.load(this.ctx.id).pipe(
+        catchError(err => of(console.error("Auto-grading refresh error:", err))),
+        first(),
```
Comment (sei-jmattson) [inline projects/gameboard-ui/src/app/game/gameboard-page/gameboard-page.component.ts:144]: first() seems unnecessary here

File: projects/gameboard-ui/src/app/game/gameboard-page/gameboard-page.component.ts [hunk 2]
```diff
@@ -205,58 +207,8 @@ export class GameboardPageComponent implements OnDestroy {
   mouseenter(e: MouseEvent, spec: BoardSpec) {
     this.hoveredItem = spec;
     spec.c = 'purple';
```
Comment (sei-jmattson) [inline projects/gameboard-ui/src/app/game/gameboard-page/gameboard-page.component.ts:209]: if cleaning this up, remove the related properties as well (L23-26) and the renderer2 constructor param.

File: projects/gameboard-ui/src/app/game/player-enroll/player-enroll.component.ts [hunk 3]
```diff
@@ -134,7 +134,10 @@ export class PlayerEnrollComponent {
 
   delete(p: Player): void {
     const sub: Subscription = this.api.delete(p.id).pipe(
-      finalize(() => sub.unsubscribe())
+      first(),
```
Comment (sei-jmattson) [inline projects/gameboard-ui/src/app/game/player-enroll/player-enroll.component.ts:137]: I'm not understanding `first()` here?

**Review by sei-jmattson — APPROVED** [review]

File: projects/gameboard-ui/src/app/game/player-enroll/player-enroll.component.ts (hunk 3 above)
Comment (sei-bstein) [reply projects/gameboard-ui/src/app/game/player-enroll/player-enroll.component.ts:137]: I'm not an rxjs expert by any stretch, but my understanding is that it's a clean way to unsubscribe from an observable that you only need one value from (like a REST call). `first()` completes the subscription after the first value is emitted. Here's, it's just a style change, but above it's necessary to make sure the subscription is disposed. I think, anyway (?)

Details on using first this way [here](https://jasonwatmore.com/post/2020/08/11/rxjs-auto-unsubscribe-from-observable-after-first-value)

Comment (sei-jmattson) [reply projects/gameboard-ui/src/app/game/player-enroll/player-enroll.component.ts:137]: Oh, good to know.  Didn't realize `first()` unsubscribed.  So you don't even need the `sub` variable anymore.

**Pull request merged**